api/src/pcapi/tasks/finance_tasks.py

```diff
@@ -0,0 +1,67 @@
+import logging
+
+from flask import current_app
+
+from pcapi import settings
+from pcapi.core.finance import api as finance_api
+from pcapi.core.finance import conf as finance_conf
+from pcapi.core.finance import models as finance_models
+from pcapi.repository import transaction
+from pcapi.routes.serialization import BaseModel
+from pcapi.tasks.cloud_task import list_tasks
+from pcapi.tasks.decorator import task
+
+
+logger = logging.getLogger(__name__)
+
+
+class GenerateInvoicePayload(BaseModel):
+    reimbursement_point_id: int
+    cashflow_ids: list[int]
+    batch_id: int
+
+
+def is_generate_invoices_queue_empty() -> bool:
+    results = list_tasks(settings.GENERATE_INVOICES_QUEUE_NAME)
+    return len(results) == 0
+
+
+@task(settings.GENERATE_INVOICES_QUEUE_NAME, "/finance/generate-store-invoice", task_request_timeout=15 * 60)
+def generate_and_store_invoice_task(payload: GenerateInvoicePayload) -> None:
+    try:
+        with transaction():
+            finance_api.generate_and_store_invoice(
+                reimbursement_point_id=payload.reimbursement_point_id,
+                cashflow_ids=payload.cashflow_ids,
+            )
+            logger.info("Generated and sent invoice", extra={"reimbursement_point_id": payload.reimbursement_point_id})
+
+            # When it's last invoice, generate and upload the invoices file
+            if current_app.redis_client.decr(finance_conf.REDIS_GENERATE_INVOICES_COUNTER) == 0:  # type: ignore [attr-defined]
+                try:
+                    batch = finance_models.CashflowBatch.query.get(payload.batch_id)
+                    path = finance_api.generate_invoice_file(batch)
+                    logger.info("Generated CSV invoices file")
+                    drive_folder_name = finance_api._get_drive_folder_name(batch)
+                    finance_api._upload_files_to_google_drive(drive_folder_name, [path])
+                    logger.info("Uploaded CSV invoices file to Google Drive")
+                    # TODO (vroullier, 2023-10-10) send mail about successful generation
+
+                except Exception as exc:  # pylint: disable=broad-except
+                    logger.exception(
+                        "Invoices generation failed",
```

Comment:
[détail] On pourrait être plus précis, ce qui ferait moins peur et aiderait (un peu) à la reprise sur erreur : 

```suggestion
                        "Failed to generate invoices CSV file",
```

Reply:
Oui c'est mieux